contracts/MarketGreedy.sol

```diff
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.15;
+pragma abicoder v2;
+import "./Market.sol";
+
+abstract contract MarketGreedy is Market {
+    // MarketID => amount of cover taken for this market
+	mapping(bytes16 => uint256) private _marketCover;
+	uint256 internal _totalCover;
+	mapping(bytes16 => bytes16) internal _mostExpensivePropositionId;
+
+    function getMarketCover(bytes16 marketId) external view returns (uint256) {
+		return _marketCover[marketId];
+	}
+
+	//TODO: Should return the dedection to total exposure to be 
+	function _payout(uint256 index, bool result) internal override {
+		require(
+			_bets[index].payoutDate < block.timestamp,
```

Comment:
I don't think this is correct, the payout date should be greater than now.

Reply:
No. This would mean than payouts could only happen before the payout date and fail afterwards.

Reply:
Yeah it looks like the `_payout` function should only be called _after_ the date has been reached